### ISSUE TITLE
-Wpointer-bool-conversion in drivers/net/ethernet/mellanox/mlx4/eq.c

### ISSUE THREAD

**Issue by shenki** [ClangBuiltLinux/linux#86]
```
drivers/net/ethernet/mellanox/mlx4/eq.c:243:11: warning: address of array 'eq->affinity_mask' will always
      evaluate to 'true' [-Wpointer-bool-conversion]
        if (!eq->affinity_mask || cpumask_empty(eq->affinity_mask))
            ~~~~~^~~~~~~~~~~~~
```

**Comment by nathanchance** [comment]
Requires `# CONFIG_CPUMASK_OFFSTACK is not set` to be triggered.

Patch sent: https://lore.kernel.org/lkml/20180921094412.15547-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Patch accepted: https://git.kernel.org/pub/scm/linux/kernel/git/davem/net.git/commit/?id=8ac1ee6f4d62e781e3b3fd8b9c42b70371427669

**Comment by nathanchance** [comment]
Merged into mainline, will be in 4.19-rc6: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8ac1ee6f4d62e781e3b3fd8b9c42b70371427669
